### ISSUE TITLE
Dataloader2 does not invoke `set_seed` if used without ReadingService

### ISSUE THREAD

**Issue by sehoffmann** [pytorch/data#1107]
### 🐛 Describe the bug

See title: My custom datapipe with `set_seed` does not get invoked when using `Dataloader2` (and calling seed(123)). This makes it impossible to fixate seeds when one does not want to use multiprocessing.

This is due to https://github.com/pytorch/data/blob/main/torchdata/dataloader2/dataloader2.py#L210

### Versions

e78ab6c9ec94f05f0a350ced7fe571f6863c20ec

**Comment by NivekT** [comment]
I think this is a good call out. The plan is to have `SingleProcessReadingService` and use that as default.

See https://github.com/pytorch/data/issues/720
